### PR TITLE
[ADF-4849] [APM-1226] - Automate boolean and number  visibility conditions

### PR DESCRIPTION
**Pull request by SilviuCPopa**
…tions

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4849
https://issues.alfresco.com/jira/browse/APM-1226

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/pages/adf/process-services/startProcessPage.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ export class StartProcessPage {
     }
 
     async clickFormStartProcessButton(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsClickable(this.formStartProcessButton);
```
Comment (adinapitul) [inline e2e/pages/adf/process-services/startProcessPage.ts]: Not needed. Click from browser-actions already contains the wait

File: e2e/process-services-cloud/form-field/visibility-condition.e2e.ts [hunk 2]
```diff
@@ -248,3 +253,161 @@ describe('Visibility conditions - cloud', () => {
...
+    });
+
+    beforeEach(async () => {
+        await navigationBarPage.navigateToProcessServicesCloudPage();
+        await appListCloudComponent.checkApsContainer();
+        await appListCloudComponent.goToApp(simpleApp);
+        await tasksCloudDemoPage.taskListCloudComponent().checkTaskListIsLoaded();
+    });
+
+    afterAll(async () => {
```
Comment (adinapitul) [inline e2e/process-services-cloud/form-field/visibility-condition.e2e.ts]: Duplicate afterAll

File: lib/testing/src/lib/core/pages/form/formFields.ts [hunk 3]
```diff
@@ -39,6 +39,12 @@ export class FormFields {
         await BrowserActions.clearSendKeys(fieldElement, value);
     }
 
+    async clickField(locator, field): Promise<void> {
+        const fieldElement: any = element(locator(field));
+        await BrowserVisibility.waitUntilElementIsClickable(fieldElement);
```
Comment (adinapitul) [inline lib/testing/src/lib/core/pages/form/formFields.ts]: Wait not needed, already done in the click method

File: e2e/process-services-cloud/form-field/visibility-condition.e2e.ts [hunk 4]
```diff
@@ -248,3 +253,155 @@ describe('Visibility conditions - cloud', () => {
...
+    const settingsPage = new SettingsPage();
+    const startTask = new StartTasksCloudPage();
+    const taskFormCloudComponent = new TaskFormCloudComponent();
+    const startProcessPage = new StartProcessPage();
+    const processCloudDemoPage = new ProcessCloudDemoPage();
+    const processDetailsCloudDemoPage = new ProcessDetailsCloudDemoPage();
+
+    const simpleApp = resources.ACTIVITI7_APPS.SIMPLE_APP.name;
+    const standaloneTaskName = StringUtil.generateRandomString(5);
+    let processName = StringUtil.generateRandomString(5);
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/visibility-condition.e2e.ts]: does this need to be `let` ?

File: e2e/process-services-cloud/form-field/visibility-condition.e2e.ts [hunk 5]
```diff
@@ -248,3 +253,155 @@ describe('Visibility conditions - cloud', () => {
...
+        await taskFormCloudComponent.formFields().checkWidgetIsVisible('Number2');
+        await expect(await taskFormCloudComponent.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+
+        await taskFormCloudComponent.formFields().setFieldValue(by.id, 'Number1', 123);
+        await expect(await taskFormCloudComponent.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        await taskFormCloudComponent.formFields().checkWidgetIsHidden('Number2');
+    });
+
+    it('[C315169] Should be able to start a process with visibility condition for number widgets', async () => {
+        const processModelWithSe = 'numbervisibilityprocess';
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/visibility-condition.e2e.ts]: this constant is used only once, I suggest just using string value in the `await startProcessPage.selectFromProcessDropdown(processModelWithSe);` line

File: e2e/process-services-cloud/form-field/visibility-condition.e2e.ts [hunk 6]
```diff
@@ -248,3 +253,155 @@ describe('Visibility conditions - cloud', () => {
...
+        const numberVisibilityTask = 'number_visibility_task';
+        processName = StringUtil.generateRandomString(5);
+
+        await processCloudDemoPage.openNewProcessForm();
+        await startProcessPage.clearField(startProcessPage.processNameInput);
+        await startProcessPage.selectFromProcessDropdown(processModelWithSe);
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/visibility-condition.e2e.ts]: The constant is never used outside this call
```suggestion
        await startProcessPage.selectFromProcessDropdown('numbervisibilityprocess');
```

File: e2e/process-services-cloud/form-field/visibility-condition.e2e.ts [hunk 7]
```diff
@@ -248,3 +253,155 @@ describe('Visibility conditions - cloud', () => {
...
+        await expect(await taskFormCloudComponent.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        await taskFormCloudComponent.formFields().checkWidgetIsHidden('Number2');
+
+        await taskFormCloudComponent.formFields().setFieldValue(by.id, 'Number1', 4);
+        await expect(await taskFormCloudComponent.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+        await taskFormCloudComponent.clickCompleteButton();
+    });
+
+    it('[C315232] Should be able to complete a process with visibility condition for number widgets', async () => {
+        const processModelWithSe = 'booleanvisibilityprocess';
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/visibility-condition.e2e.ts]: same as above, makes no sense moving a string to a constant if it's used only once

**Pull request merged**